### PR TITLE
Get rid of the workspace.Locate behavior and comms package

### PR DESCRIPTION
**Pull request by kytrinyx**
This is big in the sense that it deletes a lot of code, but the core of the PR is three small simplifications:

- the open command - https://github.com/exercism/cli/commit/20ebf1956904f21a1aa9a255f898a08aa6ac5f7b
- the submit command - https://github.com/exercism/cli/commit/693b42910ed2913a1cbef98e8d5595c419480075
- the download command - https://github.com/exercism/cli/commit/dd6b948a28b3e98eff94cc676ade19ccc30fe2ca

Once all of those were simplified, we no longer used `SolutionPath`, `IsSolutionPath`, `ResolveSolutionPath`, or `Locate`, and could delete all of the functions and their associated tests. And the entire `comms` package!

This removes the behavior of adding a numeric suffix when downloading an exercise if the exercise already exists, which is the first step in fixing https://github.com/exercism/cli/issues/661
The second step is to make the download command more robust so it doesn't clobber existing solution files: https://github.com/exercism/cli/issues/697

### CODE REVIEW

File: cmd/download.go [hunk 1]
```diff
@@ -59,21 +59,21 @@ func runDownload(cfg config.Config, flags *pflag.FlagSet, args []string) error {
...
 	if err != nil {
 		return err
 	}
-	if uuid == "" && exercise == "" {
+	if uuid == "" && slug == "" {
 		return errors.New("need an --exercise name or a solution --uuid")
 	}
 
-	var slug string
+	var param string
```
Comment (nywilken) [inline cmd/download.go]: What is param supposed to represent an exercise `iteration`?

File: cmd/download.go [hunk 2]
```diff
@@ -59,21 +59,21 @@ func runDownload(cfg config.Config, flags *pflag.FlagSet, args []string) error {
...
 	if uuid == "" {
```
Comment (nywilken) [inline cmd/download.go]: Yon can simplify this a bit
```
param := "latest"
if uuid != "" {
 param = uuid
}
```

File: cmd/download.go (hunk 1 above)
Comment (kytrinyx) [reply cmd/download.go]: param is the param to the URL that we're calling on the API. It's either the string `latest`, or an actual UUID.

File: cmd/download.go (hunk 2 above)
Comment (kytrinyx) [reply cmd/download.go]: Thanks! Yeah, that's much better. Updated.

**Pull request merged**